src/hw-3/services/user.sevice.ts

```diff
@@ -0,0 +1,60 @@
+import { NewUser, User } from '../types/user';
+import { Sequelize, DataTypes, Op } from 'sequelize';
+import { UserModel } from '../models/user';
+
+const sq = new Sequelize('postgres://postgres:genya147@localhost:5433/user_db');
+const { v1: uuidv4 } = require('uuid');
+
+export default class UserService {
+    async getUserList(query: any) {
+        const limitUserList = isNaN(query?.limit) ? null : Number(query?.limit);
+        const search = query?.search || '';
+
+        return await UserModel(sq, DataTypes)
+            .findAll({
+                raw: true,
+                where: { login:  { [Op.iLike]: `%${search}%` } },
+                order: [[ 'login', 'ASC' ]],
+                limit: limitUserList
+            });
+    }
+
+    async getUserById(id: string) {
+        return await UserModel(sq, DataTypes).findOne({ where: { id } });
+    }
+
+    async deleteUserById(id: string) {
+        return await UserModel(sq, DataTypes)
+            .update(
+                { isDeleted: true },
+                {
+                    where: { id }
+                })
+    }
+
+    async createUser(newUser: NewUser): Promise<User> {
+        const user: User = {...newUser, id: uuidv4(newUser), isDeleted: false};
```

Comment:
I think you can use sequelize model/UUID data type to generate id of the user

Reply:
Added, thanks.